packages/app-store/basecamp3/api/callback.ts

```diff
@@ -0,0 +1,83 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import prisma from "@calcom/prisma";
+
+import getAppKeysFromSlug from "../../_utils/getAppKeysFromSlug";
+import getInstalledAppPath from "../../_utils/getInstalledAppPath";
+import appConfig from "../config.json";
+import { userAgent } from "../lib/constants";
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { code } = req.query;
+  const { client_id, client_secret } = await getAppKeysFromSlug("basecamp3");
+
+  const redirectUri = WEBAPP_URL + "/api/integrations/basecamp3/callback";
+  // gets access token
+  const accessTokenResponse = await fetch(
+    `https://launchpad.37signals.com/authorization/token?type=web_server&client_id=${client_id}&redirect_uri=${redirectUri}&client_secret=${client_secret}&code=${code}`,
```

Comment:
Same comment about the hardcoded URL, just to be sure we are covered. Also, NIT but would be great if you could use `URLSearchParams` to generate que params from a JSON object, much cleaner.